test/schema-validation.test.js

```diff
@@ -105,6 +105,142 @@ test('Basic validation test', t => {
   })
 })
 
+test('Different schema per content type', t => {
+  t.plan(12)
+
+  const fastify = Fastify()
+  fastify.addContentTypeParser('application/octet-stream', {
+    parseAs: 'buffer'
+  }, async function (_, payload) {
+    return payload
+  })
+  fastify.post('/', {
+    schema: {
+      body: {
+        content: {
+          'application/json': {
+            schema: schemaArtist
```

Comment:
Why this `schema` property is needed. Are you going to add some other properties to the mediatype object?

Reply:
If we wanna solely support schema here it is not needed but I added it for two reasons

- to semantically match openapi definition and current response schema added in https://github.com/fastify/fastify/pull/4264
- and to be able to add other properties such as `description` or `example`, although fastify itself doesn't care, upstream tooling like fastify-swagger can uttitlize it